processing/src/main/java/io/druid/query/aggregation/HistogramAggregatorFactory.java

```diff
@@ -153,7 +153,14 @@ public String getFieldName()
   public byte[] getCacheKey()
   {
     byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
-    return ByteBuffer.allocate(1 + fieldNameBytes.length).put(CACHE_TYPE_ID).put(fieldNameBytes).array();
+    ByteBuffer buf = ByteBuffer
+        .allocate(1 + fieldNameBytes.length + Floats.BYTES * breaks.length)
+        .put(CACHE_TYPE_ID)
+        .put(fieldNameBytes)
+        .put((byte)0);
```

Comment:
to make sure there is a separator, otherwise it is possible to have fieldNames / breaks combinations that cause cache key collisions.